sonic_platform_base/sonic_sfp/sfputilbase.py

```diff
@@ -369,87 +374,118 @@ def read_porttab_mappings(self, porttabfile):
         first = 1
         port_pos_in_file = 0
         parse_fmt_port_config_ini = False
-
-        try:
-            f = open(porttabfile)
-        except:
-            raise
+        parse_fmt_platform_json = False
 
         parse_fmt_port_config_ini = (os.path.basename(porttabfile) == "port_config.ini")
+        parse_fmt_platform_json = (os.path.basename(porttabfile) == "platform.json")
+
+        (platform, hwsku) =  DaemonBase.get_platform_and_hwsku()
+        if(parse_fmt_platform_json):
+            ports, _ = get_port_config(hwsku, platform)
+            if not ports:
+                print('Failed to get port config', file=sys.stderr)
+                sys.exit(1)
+            else:
+                logical_list = []
+                for intf in ports.keys():
+                    logical_list.append(intf)
 
-        # Read the porttab file and generate dicts
```

Comment:
Let's keep the old code not touched, add new code on top of it in case platform.json presents, and then return.

Reply:
addressed.